### PR TITLE
ARTEMIS-4076: add JDK 19 run to CI jobs

### PR DESCRIPTION
**Pull request by gemmellr**
Fixes a couple of issues causing test failures or hangs, and adds a Java 19 job to the CI runs on push/PR.

### CODE REVIEW

File: artemis-unit-test-support/src/main/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ private boolean isExpectedThread(Thread thread) {
          return true;
       } else if (threadName.contains("Attach Listener")) {
          return true;
-      } else if ((javaVendor.contains("IBM") || isSystemThread) && threadName.equals("process reaper")) {
+      } else if ((javaVendor.contains("IBM") || isSystemThread) && (threadName.equals("process reaper") || threadName.startsWith("process reaper (pid"))) {
```
Comment (clebertsuconic) [inline artemis-unit-test-support/src/main/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java:247]: did you mean the (pid) in the string?

Comment (clebertsuconic) [reply artemis-unit-test-support/src/main/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java:247]: yes... I think you just do startsWith("process reaper");


As the change on the idk is 

process reaper (pid " + pid + ")");

Comment (gemmellr) [reply artemis-unit-test-support/src/main/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java:247]: It isnt always in the name, it comes and goes.

Comment (gemmellr) [reply artemis-unit-test-support/src/main/java/org/apache/activemq/artemis/utils/ThreadLeakCheckRule.java:247]: But yes we could use 'starts with' for both if you prefer.

**Pull request merged**